client/components/EditPostForm.jsx

```diff
@@ -0,0 +1,175 @@
+const ImageIcon = require('@phosphor-icons/react').Image // Alias for Phosphor Icons "Image", since it shares the same name as NextUI's "Image"
+const { getIndividualPost, getAllPhrases, getAllStockImages } = require('../dataHelper.js')
+
+/**
+ * Renders the create post form.
+ * @param {object} props - Unused.
+ */
+function EditPostForm (props) {
+    const { id } = useParams();
+    const navigateTo = useNavigate()
+    const [cookies] = useCookies(["token"])
+    const [modalState, setModalState] = React.useState(true) //* true renders phrases, false renders photos
+    const [isFormDisabled, setIsFormDisabled] = React.useState(false)
+    const [phrase, setPhrase] = React.useState(undefined)
+    const [picture, setPicture] = React.useState(null)
+    const [stockPhrases, setStockPhrases] = React.useState([])
+    const [stockImages, setStockImages] = React.useState([])
+
+    const {isOpen, onOpen, onOpenChange} = useDisclosure()
+
+    const imagePath = "/pictures/stock_images/"
+
+    const selectPhrase = (phraseString) => {
+        setPhrase(phraseString)
+        onOpenChange()
+    }
+    
+    const selectPicture = (imageURL) => {
+        setPicture(imageURL)
+        onOpenChange()
+    }
+
+    // get the post data and autopopulate the form
+    React.useEffect(() => {
+        onStart();
+    }, []);
+
+    async function onStart() {
+        await getIndividualPost(cookies.token, id).then((postData) => {
+            setPhrase(postData[0].content);
+            setPicture(postData[0].picture);
+        })
+        await getAllStockImages(cookies.token).then((imageData) => {
+            setStockImages(imageData);
+        })
+    }
+
+    return (
+        <div className="grid justify-items-center mt-6">
+            <Card className="flex-auto w-1/2">
+                <CardHeader>
+                    <Button
+                        isDisabled={isFormDisabled}
+                        size="sm"
+                        radius="full"
+                        startContent={<ArrowBendUpLeft size={16}/>}
+                        onClick={() => {
+                            navigateTo("/feed")
+                        }}
+                    >
+                        Back to Feed
+                    </Button>
+                </CardHeader>
+                <CardBody>
+                    <Textarea
+                        isReadOnly
+                        isDisabled={isFormDisabled}
+                        className="cursor-pointer"
+                        minRows={8}
+                        size="lg"
+                        radius="none"
+                        value={phrase}
+                        defaultValue={phrase}
+                        description="Click the field above to select a predefined phrase for your post."
+                        onClick={async () => {
+                            if(!isFormDisabled) {
+                                setStockPhrases(await getAllPhrases(cookies.token))
```

Comment:
surround this with an `if` checking if the stock phrases list is empty; if so, you can get the phrases

otherwise, we're making a database call everytime we click on the thing